src/python/pants/rules/core/test_test.py

```diff
@@ -121,41 +122,35 @@ def test_stderr(self):
       success=False,
     )
 
-  def test_coordinator_python_test(self):
-    addr = Address.parse("some/target")
-    target_adaptor = PythonTestsAdaptor(type_alias='python_tests')
-    with self.captured_logging(logging.INFO):
-      result = run_rule(
-        coordinator_of_tests,
-        rule_args=[
-          HydratedTarget(addr, target_adaptor, ()),
-          UnionMembership(union_rules={TestTarget: [PythonTestsAdaptor]}),
-          AddressProvenanceMap(bfaddr_to_spec={}),
-        ],
-        mock_gets=[
-          MockGet(
-            product_type=TestResult,
-            subject_type=PythonTestsAdaptor,
-            mock=lambda _: TestResult(status=Status.FAILURE, stdout='foo', stderr=''),
-          ),
-        ],
+  def run_coordinator_of_tests(
```

Comment:
This function was added to DRY the coordinator tests. I developed through this flow:
1) Get all original tests working with this function, without changes to src
2) Get all original tests working with this function, with changes to src
3) Add new test `test_coordinator_empty_sources`

I also renamed some of the tests to have a consistent naming scheme of `test_coordinator_x`.